middleware/interceptors/server/logger.go

```diff
@@ -26,6 +26,11 @@ func LoggerGRPCInterceptor(deps loggerInterceptorDeps) grpc.UnaryServerIntercept
 		resp, err = handler(ctx, req)
 		if logLevel := deps.Config.Get(confkeys.MiddlewareLogLevel); logLevel.IsSet() {
 			level := log.ParseLevel(logLevel.String())
+			if err != nil {
+				if logErrorLevel := deps.Config.Get(confkeys.MiddlewareLogErrorLevel); logErrorLevel.IsSet() {
+					level = log.ParseLevel(logErrorLevel.String())
```

Comment:
In this case we should take the higher of the 2.
Example: 
By default someone configured to log on Info level, but forgot and configured onError to be on trace level.
In this case when we encounter an actual error it will be logged using the trace level and it will not be logged.